### DISCUSSION TITLE
Event Title

### DISCUSSION THREAD

**Discussion by PSanilP** [vkurko/calendar#148] [General]
The event title always starts with the time 
Like so
10:00 AM Some Event Title

Is there a way to hide the 10:00 AM
?
I checked the titleHTML property, but the time still shows

**Comment by PSanilP** [comment]
ok sorted it
           eventTimeFormat: function () { return ""; },
                eventTextColor: "#000000",

eventTextColor is not working though

**Comment by vkurko** [comment]
You can completely override event rendering using the [eventContent](https://github.com/vkurko/calendar#eventcontent) option.

In this [demo](https://jsbin.com/banatudowa/edit?js,output), the `eventTextColor` option seems to work fine, and I also showed you how to use the `eventContent` option.

**Reply by PSanilP** [reply]
https://codepen.io/psanilp/pen/KKGRWZo
this is what I tried with eventTextColor
Maybe the problem is with Vue

**Reply by vkurko** [reply]
`eventTextColor` was introduced in [v1.1.0](https://github.com/vkurko/calendar/blob/master/CHANGELOG.md#110) and you're using v0.19.0.

**Reply by PSanilP** [reply]
Ok. Works

**Reply by PSanilP** [reply]
Another query. in the example https://codepen.io/psanilp/pen/KKGRWZo
if I update the event data from backend, how do I force calendar to refresh?
 datesSet: function (info) {
              // send a post request to my api, which will then return events for new startdate+42 .
//. when data comes back, what to do to refresh?
                   refetchEvents()
// is this correct?
                }

**Reply by vkurko** [reply]
Please see #129

**Reply by PSanilP** [reply]
is there an example? This EventSources thing is too complicated
Why cant I update the events property and call a re-render function of some sort?

**Reply by vkurko** [reply]
Of course you can. If you are overwriting the `events` option, then you do not need to call `refetchEvents()`. The calendar should be redrawn immediately after updating the option.

**Comment by PSanilP** [comment]
Somehow changing the events property and calling refetchEvents() is not working
I am using vueJS by the way
v-on:click="xy.ec.events=xz.myData2; xy.ec.refetchEvents();"
this doesnt work

this works
v-on:click="xy.ec.removeEventById('b2');"

BUT.. you don't have a method removeAllEvents()
That would be very convenient
I am not able to set the events array to [] without looping through the Ids

**Reply by vkurko** [reply]
Try `ec.setOption('events', xz.myData2)`.

`refetchEvents()` doesn't make any sense in your case.

**Comment by PSanilP** [comment]
Brilliant. That works! Thank you for prompt feedback
Was getting frustrated.
In my use case, I had many conditions on the screen for the user to define. Based on choice, the ajax call to api would return new event array. Was not able to update the calendar even though the vue variable was changing.
Something to do with the proxy object that vue was creating. Anyway, thanks again for the help

**Comment by PSanilP** [comment]
Its me again. The calendar works, but recently I added 'week view' to it.
The events do not appear! What is the event I need to look for?
![image](https://github.com/vkurko/calendar/assets/86600502/52405ae8-3561-4a09-8654-2fc843664a05)
![image](https://github.com/vkurko/calendar/assets/86600502/d480bd61-3439-467a-b98f-3d15aa5086d8)

**Reply by PSanilP** [reply]
I sorted this out.. it was because of Time hour range not valid.
A query though.. how do you set the value to be displayed in the event title? In the week view, it only shows time, not the 'client name' as the height is not enough. It only shows if the event is 1 hour long. How do I 'stretch' the space between hours?

**Comment by PSanilP** [comment]
Sorted.. thanks for the documentation :)
 slotDuration: '00:15:00', slotMinTime: '06:30:00', slotMaxTime: '21:30:00',
